dcmgr/lib/dcmgr/models/virtual_data_center.rb

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+
+module Dcmgr::Models
+  class VirtualDataCenter < AccountResource
+    taggable 'vdc'
+
+    one_to_one :virtual_data_center_spec
+    alias :vdc_spec :virtual_data_center_spec 
+
+    one_to_many :virtual_data_center_instance
+    alias :vdc_instances :virtual_data_center_instance
+
+    many_to_many :instances, :class=>Instance, :join_table=>:virtual_data_center_instances, :right_key=>:instance_id, :right_primary_key=>:id do |ds|
+      # "SELECT `instances`.* FROM `instances` INNER JOIN `virtual_data_center_instances` ON 
+      # ((`virtual_data_center_instances`.`instance_id` = `instances`.`id`) AND (`virtual_data_center_instances`.`virtual_data_center_id` = 56))
+      # WHERE (terminated_at IS NULL OR terminated_at >= '2015-07-29 05:59:15')"
+      ds.alives_and_termed
+    end
+
+    def self.entry_new(account)
+      raise ArgumentError, "The account parameter must be an Account. Got '#{account.class}'" unless account.is_a?(Account)
+      vdc = self.new
+      vdc.account_id = account.canonical_uuid
+      vdc.save
+      vdc
+    end
+
+    def add_virtual_data_center_spec(type, spec, spec_file = nil)
+      # Mash is passed in some cases.
+      raise ArgumentError, "The type parameter must be a String. Got '#{type.class}'" if !type.is_a?(String)
+      raise ArgumentError, "The spec parameter must be a String. Got '#{spec.class}'" if !spec.is_a?(String)
```

Comment:
While it's very good to do these error checks, I think it would be a good idea to make a helper method for them. We can do the same thing OpenVNet did here.

https://github.com/axsh/openvnet/blob/develop/vnet/lib/ext/kernel.rb